gitlabform/gitlabform/core.py

```diff
@@ -424,15 +434,27 @@ def process_secret_variables(self, project_and_group, configuration):
     @if_in_config_and_not_skipped
     def process_group_secret_variables(self, group, configuration):
         logging.debug("Group secret variables BEFORE: %s", self.gl.get_group_secret_variables(group))
+
         for secret_variable in sorted(configuration['group_secret_variables']):
             logging.info("Setting group secret variable: %s", secret_variable)
 
+            if "delete" in configuration['group_secret_variables'][secret_variable]:
+                key = configuration['group_secret_variables'][secret_variable]["key"]
+                if configuration['group_secret_variables'][secret_variable]["delete"]:
+                    logging.debug(f"Deleting {secret_variable}: {key} in group {group}")
+                    try:
+                        self.gl.delete_group_secret_variable(group, key)
+                    except:
+                        logging.warn(f"Could not delete variable {key} in group {group}")
+                    return
+
             try:
                 self.gl.put_group_secret_variable(group, configuration['group_secret_variables'][secret_variable])
             except NotFoundException:
                 self.gl.post_group_secret_variable(group, configuration['group_secret_variables'][secret_variable])
 
         logging.debug("Groups secret variables AFTER: %s", self.gl.get_group_secret_variables(group))
+        logging.info("gRoups secret variables AFTER: %s", self.gl.get_group_secret_variables(group))
```

Comment:
This line almost duplicates the one above so I would delete it. If someone needs to see group variables before and after run the they should use `-d` to enable the debug output.

Reply:
yeah, sorry left that in when debugging